_drafts/STMViewer.md

```diff
@@ -0,0 +1,161 @@
+---
+# This is a template for Interrupt posts. Use previous, more recent posts from the _posts/
+# directory for more inspiration and patterns.
+#
+# When submitting a post for publishing, submit a PR with the post in the _drafts/ directory
+# and it will be moved to _posts/ on the date of publish.
+#
+# e.g.
+# $ cp _drafts/_template.md _drafts/my_post.md
+#
+# It will now show up in the front page when running Jekyll locally.
+
+title: Visualizing embedded data made easy
+description: 
+    "A brief introduction to STMViewer software use cases in debugging embedded targets"
+author: piotrw
+---
+
+<!-- excerpt start -->
+
+If you've ever wanted to plot data acquired on your embedded target, this article is for you. It explores common use cases for real-time data visualization using STMViewer. Say goodbye to manual, time-consuming, and error-prone data collection and display methods to speed up your debugging process.
+
+<!-- excerpt end -->
+
+{% include newsletter.html %}
+
+{% include toc.html %}
+
+## What exactly can it visualize?
+
+STMViewer consists of two modules which serve a slightly different purpose: 
+
+1. Variable Viewer - a module that reads variables' values directly from RAM using ST-Link SWD (only SWDIO, SWCLK, and GND connection is needed). It is asynchronous, which means it samples the memory with the time period set on the PC. It is completely non-intrusive and can log many variables at once. There are some trade-offs though. The first one is that logged variables' addresses have to be constant, meaning they have to be globals. Moreover, with more aggressive optimization some of the variables may get optimized and fail to be logged.
+
+2. Trace Viewer - a module that reads and parses SWO (Serial Wire Viewer) data using an ST-Link. To use it, you need to connect SWDIO, SWCLK, SWO, and GND. This tool operates synchronously, meaning it visualizes data points at the rate they are produced on the target. A data point can be either a special single-byte flag used to create digital plots or any value that fits within up to 4 bytes, as in the case of 'analog' plots. It incurs minimal performance overhead, involving just a single register write per plot point, as serialization and timestamping are handled by the ITM peripheral. Trace Viewer can be employed for profiling specific parts of the codebase or visualizing signals that are too fast for the asynchronous Variable Viewer. Its main limitations are the maximum allowed SWO pin baudrate of the ST-Link and the fact that the ITM peripheral is supported only on Cortex M3/M4/M7/M33 cores. 
```

Comment:
Wondering if these would flow better as sections like `### Variable Viewer`. As it stands they're both a bit long for a numbered list.